### PR TITLE
test: tidy up new RTL tests - FE-6549

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour

Tidy up earlier tests converted to RTL

### Current behaviour

Earlier converted RTL tests may not be aligned with our current standards

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions

N/A

### CODE REVIEW

**Review by mihai-albu-sage — DISMISSED** [review]

**Review by tomdavies73 — COMMENTED** [review]
@DipperTheDan  I've noticed that there are still some tests in the codebase which use `toHaveStyle` testing against token values. 

It would be good to convert these to use `toHaveStyleRule`, as currently the majority of these tests will return false positives 👍

File: src/components/button-bar/button-bar.test.tsx [hunk 1]
```diff
@@ -61,10 +61,11 @@ describe("When ButtonBar children are Button components", () => {
...
     buttonChildren.forEach((button) => {
-      expect(button).toHaveStyle({
-        fontSize: "var(--fontSizes100)",
-        minHeight: "40px",
-      });
+      expect(button).toHaveStyleRule("min-height", "40px");
+    });
+
+    buttonChildren.forEach((button) => {
+      expect(button).toHaveStyleRule("font-size", "var(--fontSizes100)");
```
Comment (tomdavies73) [inline src/components/button-bar/button-bar.test.tsx:68]: **question**: are both `toHaveStyleRule` checks meant to be within the `forEach` here?

Comment (DipperTheDan) [reply src/components/button-bar/button-bar.test.tsx:68]: Ah my mistake. The first one was meant to be an ordinary `toHaveStyle` as it doesn't use tokens. I'll get it amended.

**Review by tomdavies73 — DISMISSED** [review]
LGTM @DipperTheDan, up to you if you want to combine commits for tests where you've done general tidying up + `toHaveStyle` -> `toHaveStyleRule` 👍

**Review by mihai-albu-sage — DISMISSED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by mihai-albu-sage — APPROVED** [review]

**Pull request merged**